src/Statement/Grouper/FilteringStatementGrouper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use Wikibase\DataModel\Services\Statement\Filter\StatementFilter;
+use Wikibase\DataModel\Statement\Statement;
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class FilteringStatementGrouper implements StatementGrouper {
+
+	/**
+	 * @var StatementList[]
+	 */
+	private $groups = array();
+
+	/**
+	 * @var StatementFilter[] An associative array, mapping statement group identifiers to filters.
+	 */
+	private $filters = array();
+
+	/**
+	 * @var string
+	 */
+	private $defaultGroupIdentifier = 'statements';
+
+	/**
+	 * @param array $filters An associative array, mapping statement group identifiers to either
+	 *  filters, or to null for the default group.
```

Comment:
filters should link to the StatementFilter interface, so it is clear how to provide a filter